File: LeetCode/Simulation/JS/Leetcode1768-merge-strings-alternately.js
var mergeAlternately = function (word1, word2) {
  const n = Math.min(word1.length, word2.length);
  let ans = "";
  for (let i = 0; i < n; i++) {
    ans += word1[i];
    ans += word2[i];
  }
  if (n != word1.length) ans += word1.substr(n);
  if (n != word2.length) ans += word2.substr(n);
  return ans;
};

File: LeetCode/Simulation/Golang/Leetcode1768-merge-strings-alternately.go
package sbgo

import "strings"

func mergeAlternately(word1 string, word2 string) string {
	min := func(i, j int) int {
		if i < j {
			return i
		}
		return j
	}
	n, ans := min(len(word1), len(word2)), strings.Builder{}
	for i := 0; i < n; i++ {
		ans.WriteByte(word1[i])
		ans.WriteByte(word2[i])
	}
	if n != len(word1) {
		ans.Write([]byte(word1[n:]))
	}
	if n != len(word2) {
		ans.Write([]byte(word2[n:]))
	}
	return ans.String()
}
